src/IPC/QueueTx.cc

```diff
@@ -1,22 +1,97 @@
 #include <iostream>
 #include "IPC.h"
 
+#include "Queue.h"
+
+#include <sys/stat.h>   /* S_IRWXU S_IRWXG */
+#include <sys/ipc.h>    /* IPC_CREAT */
+#include <sys/msg.h>    /* msgget() */
+#include <string.h>     /* strerror() */
+
+char readBuf[BUFFERSIZE];
+
 void QueueTx::send(void) {
     this->setupQueueTx();
 
-    this->fileSizeTx();
-
     this->queueTx();
 }
 
-void QueueTx::setupQueueTx(void) const {
-    ;
+void QueueTx::setupQueueTx(void) {
+    key_t key;
+
+    /* ftok - convert a pathname and a project identifier to a System V IPC key */
+    key = ftok(PathName, ProjectId);
+    if (key < 0) {
+        throw std::runtime_error(strerror(errno));
+    }
+
+    /* msgget - get a System V message queue identifier */
+    this->qid = msgget(key, S_IRWXU | S_IRWXG | IPC_CREAT);
+    if (qid < 0) {
+        throw std::runtime_error(strerror(errno));
+    }
 }
 
-void QueueTx::fileSizeTx(void) const {
-    ;
+void QueueTx::queueTx(void) {
+    // char readBuf[BUFFERSIZE_QUEUE];
+    int readBytes = 0;
+    long n = 1; 
+    int status = 0; 
+
+    while(1) {
+        /* reading from readFile */
+        readBytes = this->readFile->read(readBuf, BUFFERSIZE).gcount();
+        if ( (this->readFile->rdstate() & std::ifstream::badbit) != 0 ) {
+            throw std::runtime_error("istream::read error");
+        }
+
+        /* if EOF reached, send last part and break loop */
+        if (this->readFile->eof()) {
+            /* build the message */
+            queuedMessage msg;
+            msg.index = n;
+            msg.endIndex = n;
+            msg.sizeMessage = readBytes;
+            strcpy(msg.payload, readBuf);
```

Comment:
If you want to keep this copy I would highly suggest you to replace strcpy by mem cpy, because if the file is a binary that contains /0 the strcpy will think that it is the end of the file and won't copy it, I couldn't try because your program doesn't launch for me (I'll provide you a screenshot of my execution).